src/reducers/forecastReducer.js

```diff
@@ -0,0 +1,25 @@
+import { FETCH_FORECAST } from "../actions";
+
+export const forecastReducer = function (state = [], action) {
+  switch (action.type) {
+    case FETCH_FORECAST:
+      if (
+        action.payload.data &&
+        !state.some((forecast) => forecast.id === action.payload.data.city.id)
```

Comment:
This could use a comment as it seems like something is out of place - is this so that a city does not show up twice? I think you'd prefer to prevent the request altogether.